### PR TITLE
Freemabd1/rw 401 Updating UX for cohort builder search.

### PR DESCRIPTION
**Pull request by freemabd1**

### CODE REVIEW

File: ui/src/app/cohort-search/redux/store/initial.ts [hunk 1]
```diff
@@ -9,8 +9,8 @@ export const initialState = fromJS({
   entities: {
     searchRequests: {
       [SR_ID]: {
-        includes: ['include0'],
-        excludes: ['exclude0'],
+        includes: [],
```
Comment (johnstill) [inline ui/src/app/cohort-search/redux/store/initial.ts]: Since we axed these we should also remove their initial implementations (the contents of `groups` in this file).  We'll need to also check the tests / components to make sure everything still works.

File: ui/src/app/cohort-search/search-group-list/search-group-list.component.html [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-    </button>
-  </div>
 </div>
 
 <!-- .size, not .length => this is an immutable.js list -->
-<div *ngIf="(groups$ | async)?.size > 0" class="search-groups">
+<div class="search-groups">
   <div *ngFor="let group of (groups$ | async); let i = index">
-    <div *ngIf="i > 0" class="text-center font-weight-bold">AND</div>
+    <!--<div *ngIf="i > 0" class="text-center font-weight-bold">AND</div>-->
```
Comment (johnstill) [inline ui/src/app/cohort-search/search-group-list/search-group-list.component.html]: If this is dead code let's remove it.  Also it looks like we're no longer using the `i` from `i = index`, we can remove that part as well.

File: ui/src/app/cohort-search/cohort-search.module.ts [hunk 3]
```diff
@@ -52,5 +54,6 @@ const routes: Routes = [{
   ]
 })
 export class CohortSearchModule {
-  constructor(store: ConfigureStore) {}
+  constructor(store: ConfigureStore) {
```
Comment (johnstill) [inline ui/src/app/cohort-search/cohort-search.module.ts]: Lets revert this change.

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.html [hunk 4]
```diff
@@ -7,7 +7,7 @@
       </app-search-group-list>
     </div>
 
-    <div class="col-xl-4 col-lg-6" id="exclude-groups">
+    <div *ngIf="(initShowChart$ | async)" class="col-xl-4 col-lg-6" id="exclude-groups">
```
Comment (johnstill) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.html]: Lets put this `*ngIf` direclty on `app-search-group-list` rather than the column div.  And since we have `initShowChart$` here anyways, let's put the same check on the `app-overview` component in this template, and remove it from the `overview` component.  That way this logic stays together at the same level.

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.ts [hunk 5]
```diff
@@ -38,6 +38,7 @@ export class CohortSearchComponent implements OnInit, OnDestroy {
   @select(chartData) chartData$: Observable<List<any>>;
   @select(isRequstingTotal) isRequesting$: Observable<boolean>;
   @select(activeCriteriaType) criteriaType$: Observable<string>;
+  @select(s => s.get('initShowChart')) initShowChart$: Observable<boolean>;
```
Comment (johnstill) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.ts]: immutable.js methods that involve getting / reading values usually take a `default` parameter that is returned if the attribute can't be found; here we may want to consider `s.get('initShowChart', true)`.

On the one hand, if the state is corrupted somehow then this section of the app won't crash.  On the other hand, a default value here could hide corruption by hiding the crash.

File: ui/src/app/cohort-search/redux/reducer.ts [hunk 6]
```diff
@@ -44,6 +45,7 @@ import {
   RESET_STORE,
   RootAction,
 } from './actions/types';
+import {activeRole} from './store/selectors';
```
Comment (johnstill) [inline ui/src/app/cohort-search/redux/reducer.ts]: Lets move this up with the other imports from './store' (a few lines up).

File: ui/src/app/cohort-search/redux/store/initial.ts (hunk 1 above)
Comment (freemabd1) [reply ui/src/app/cohort-search/redux/store/initial.ts]: Done.

File: ui/src/app/cohort-search/search-group-list/search-group-list.component.html (hunk 2 above)
Comment (freemabd1) [reply ui/src/app/cohort-search/search-group-list/search-group-list.component.html]: Done.
